modules/samples/firealarm/component/ui/src/main/resources/jaggeryapps/devicemgt/app/units/cdmf.unit.device.type.firealarm.type-view/public/js/download.js

```diff
@@ -0,0 +1,195 @@
+    $('label[for=deviceName]').remove();
+    $('.control-group').removeClass('success').removeClass('error');
+    $(modalPopupContent).html('');
+    $(modalPopup).hide();
+}
+
+/*
+ * DOM ready functions.
+ */
+$(document).ready(function () {
+    attachEvents();
+});
+
+function attachEvents() {
+    /**
+     * Following click function would execute
+     * when a user clicks on "Download" link
+     * on Device Management page in WSO2 DC Console.
+     */
+    $("a.download-link").click(function () {
+        var sketchType = $(this).data("sketchtype");
+        var deviceType = $(this).data("devicetype");
+        var downloadDeviceAPI = "/devicemgt/api/devices/sketch/generate_link";
+        var payload = {"sketchType": sketchType, "deviceType": deviceType};
+        $(modalPopupContent).html($('#download-device-modal-content').html());
+        showPopup();
+        var deviceName;
+        $("a#download-device-download-link").click(function () {
+            $('.new-device-name').each(function () {
+                if (this.value != "") {
+                    deviceName = this.value;
+                }
+            });
+            $('label[for=deviceName]').remove();
+            if (deviceName && deviceName.length >= 4) {
+                payload.deviceName = deviceName;
+                invokerUtil.post(
+                        downloadDeviceAPI,
+                        payload,
+                        function (data, textStatus, jqxhr) {
+                            doAction(data);
+                        },
+                        function (data) {
+                            doAction(data);
+                        }
+                );
+            } else if (deviceName) {
+                $('.controls').append('<label for="deviceName" generated="true" class="error" ' +
+                                      'style="display: inline-block;">Please enter at least 4 ' +
+                                      'characters.</label>');
+                $('.control-group').removeClass('success').addClass('error');
+            } else {
+                $('.controls').append('<label for="deviceName" generated="true" class="error" ' +
+                                      'style="display: inline-block;">This field is required.' +
+                                      '</label>');
+                $('.control-group').removeClass('success').addClass('error');
+            }
+        });
+
+        $("a#download-device-cancel-link").click(function () {
+            hidePopup();
+        });
+
+    });
+}
+
+function downloadAgent() {
+    var deviceName = "";
+    $('.new-device-name').each(function () {
+        if (this.value != "") {
+            deviceName = this.value;
+        }
+    });
+    var deviceType = "";
+    $('.deviceType').each(function () {
+        if (this.value != "") {
+            deviceType = this.value;
+        }
+    });
+    var sketchType = "";
+    $('.sketchType').each(function () {
```

Comment:
Be consistent on using quotes for string definitions. use double quotes as the standard.